diff --git a/src/common/tusb_compiler.h b/src/common/tusb_compiler.h
index d62ddf9..63bad9f 100644
--- a/src/common/tusb_compiler.h
+++ b/src/common/tusb_compiler.h
@@ -83,7 +83,7 @@
  * - ##__VA_ARGS__ is used to deal with 0 paramerter (swallows comma)
  *------------------------------------------------------------------*/
 #if !defined(__CCRX__)
-#define TU_ARGS_NUM(...)   _TU_NARG(_0, ##__VA_ARGS__, _RSEQ_N())
+#define TU_ARGS_NUM(...)   _TU_NARG(_0 __VA_OPT__(,) ##__VA_ARGS__, _RSEQ_N())
 #else
 #define TU_ARGS_NUM(...)   _TU_NARG(_0, __VA_ARGS__, _RSEQ_N())
 #endif
